raft/raft.go

```diff
@@ -766,6 +766,40 @@ func (r *raft) becomeLeader() {
 	r.logger.Infof("%x became leader at term %d", r.id, r.Term)
 }
 
+func (r *raft) hup(t CampaignType) {
```

Comment:
can you do the cleanup in a separate commit to make it easier for review?

Reply:
Done, the commit are split into 4, each is summarized as its title.